### PR TITLE
Switch emails to templates and provide both richer html/text-only versions

### LINKED ISSUE #1548: Additional Content in Report Emails
Currently when I receive an email about a video, the subject line is `[PeerTube] Received a video abuse` and contains a link to the video that has been reported. Could additional information be added to this email such as title of video, user who posted video, user who reported video, contents of report, and perhaps a link to the `/admin/moderation/video-abuses/list` URL on the specific instance, to faster deal with reports?

### LINKED ISSUE #2140: Please add HTML to email with long links
**What happened?**

A friend tried to use the password reset feature. They have indeed received an email with a link. After clicking on the link, the reset form would not work and the error message said “Invalid verification string”.

As the full link in the email is very long, I assume either their email client did not parse the link fully, or they failed to copy/paste the entire link and thus missed the last part of the URL which is contains the verification string.

**What do you expect to happen instead?**

Get an email with « Visit the reset form » as clickable link to the full URL. In short provide an HTML version. :)

**Steps to reproduce:**

1. Ask to reset a password.

2. Don't copy the full link in a browser.

3. Try to change the password.

**Additional information**
* PeerTube version or URL: v1.4.1

### PR DESCRIPTION
**Pull request by rigelk**
As outlined by #2140, there is a need for better handling of links, and while HTML is not the only option, it certainly can improve UX of mails that call to action (with the addition of a button, see one below).

Example of a report mail notification

I took the opportunity to add some info in the abuse report notification of #1548; more info and a more polished version of each email should be part of a larger analysis of all our email templates. I tried to rephrase some and add call to action buttons when using one was the obvious choice.

Text versions are also automatically generated from the HTML template. If necessary, this behaviour can be overriden, but I made sure the templates convert well enough. i.e. for the same mail:

<details>
<summary>mail content, click to expand</summary>
<pre>
 ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌  A VIDEO IS PENDING MODERATION 
 

Hi,

localhost:9000 received an abuse report for the video "Test video 4
[http://localhost:9000/videos/watch/548bad51-5806-4507-bb08-0b223e0a8c28]" by
Main root channel (root_channel@localhost:9000), published at 2020-04-15
15:51:52. 

The reporter, Test User 1 (root@localhost:9000), cited the following reason:

> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
incididunt ut labore et dolore magna aliqua.

Cheers,
PeerTube 

View report #83
[http://localhost:9000/admin/moderation/video-abuses/list?search=%2383] View in
your notifications [https://localhost:9000/my-account/notifications]
Manage your notification preferences in your profile
[https://localhost:9000/my-account/settings#notifications]
PeerTube © 2015-2020 [https://github.com/Chocobozzz/PeerTube] PeerTube
Contributors [https://github.com/Chocobozzz/PeerTube/blob/master/CREDITS.md]
</pre>
</details>

closes #1548
closes #2140

---

Notes about the implementation:
- the HTML is responsive and based on https://tedgoas.github.io/Cerberus - it is crude but should be as compatible as it can be
- the `email-templates` library used under the hood can be plugged to an `i18n-node` module, which should ease translation of emails
- the template language used is pug, which allows to keep mail templates as minimal as possible for each email type. It also has the good taste of being able to use mixins and js libraries, though I didn't use much of that so far

Do you see something missing? Would you rephrase some sentences?

### CODE REVIEW

File: server/lib/emailer.ts [hunk 1]
```diff
@@ -262,35 +242,45 @@ class Emailer {
...
+      template: 'video-abuse-new',
       to,
-      subject: CONFIG.EMAIL.SUBJECT.PREFIX + 'Received a video abuse',
-      text
+      subject: `New video abuse report from ${parameters.reporter}`,
+      locals: {
+        videoUrl,
+        videoAbuseUrl,
+        videoCreatedAt: (new Date(parameters.videoAbuseInstance.Video.createdAt)).toISOString().replace(/T/, ' ').replace(/\..+/, ''),
+        videoPublishedAt: (new Date(parameters.videoAbuseInstance.Video.publishedAt)).toISOString().replace(/T/, ' ').replace(/\..+/, ''),
```
Comment (Chocobozzz) [inline server/lib/emailer.ts:277]: Use `toLocaleString()` no?

File: server/lib/emailer.ts [hunk 2]
```diff
@@ -474,14 +426,39 @@ class Emailer {
...
+            },
+            locals: {
+              WEBSERVER,
+              EMAIL: CONFIG.EMAIL,
+              text: options.text
+            }
+          },
+          options
+        ) as SendEmailOptions)
+        .then(logger.log)
```
Comment (Chocobozzz) [inline server/lib/emailer.ts:460]: `logger.log`? Not `logger.info`?

File: server/lib/emails/common/base.pug [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+  // Desktop Outlook chokes on web font references and defaults to Times New Roman, so we force a safe fallback font.
+  //if mso
+    style.
+      * {
+      font-family: sans-serif !important;
+      }
+  // All other clients get the webfont reference; some will render the font and others will silently fail to the fallbacks. More on that here: http://stylecampaign.com/blog/2015/02/webfont-support-in-email/
+  // [if !mso] <!
+  // insert web font reference, eg:
+  link(href='https://fonts.googleapis.com/css?family=Roboto:400,700' rel='stylesheet' type='text/css')
```
Comment (Chocobozzz) [inline server/lib/emails/common/base.pug:43]: Nope

Comment (rigelk) [reply server/lib/emails/common/base.pug:43]: Damn ><

File: server/controllers/api/videos/abuse.ts [hunk 4]
```diff
@@ -128,14 +131,15 @@ async function reportVideoAbuse (req: express.Request, res: express.Response) {
       await sendVideoAbuse(reporterAccount.Actor, videoAbuseInstance, videoInstance, t)
     }
 
-    auditLogger.create(reporterAccount.Actor.getIdentifier(), new VideoAbuseAuditView(videoAbuseInstance.toFormattedJSON()))
+    videoAbuse = videoAbuseInstance.toFormattedJSON()
```
Comment (Chocobozzz) [inline server/controllers/api/videos/abuse.ts:134]: Just rename to `videoAbuseJSON` please

File: server/lib/activitypub/process/process-flag.ts [hunk 5]
```diff
@@ -36,24 +37,28 @@ async function processCreateVideoAbuse (activity: ActivityCreate | ActivityFlag,
...
         videoAbuseInstance.Video = video
+        videoAbuseInstance.Account = reporterAccount
 
         logger.info('Remote abuse for video uuid %s created', flag.object)
 
         return videoAbuseInstance
       })
 
-      Notifier.Instance.notifyOnNewVideoAbuse(videoAbuse)
+      const videoAbuse = videoAbuseInstance.toFormattedJSON()
```
Comment (Chocobozzz) [inline server/lib/activitypub/process/process-flag.ts:59]: Same here

**Pull request merged**